includes/subscriptions/class-wc-payments-subscriptions-event-handler.php

```diff
@@ -119,7 +119,18 @@ public function handle_invoice_paid( array $body ) {
 		}
 
 		if ( $order->needs_payment() ) {
+			/*
+			 * Temporarily place the subscription on-hold to imitate the normal subscription renewal flow.
+			 * This ensures the downstream effects take place, e.g. a payment status order note is added and the
+			 * 'woocommerce_subscription_payment_complete' action is fired.
+			 */
+			remove_action( 'woocommerce_subscription_status_on-hold', [ $this->subscription_service, 'suspend_subscription' ] );
+			$subscription->update_status( 'on-hold' );
+			add_action( 'woocommerce_subscription_status_on-hold', [ $this->subscription_service, 'suspend_subscription' ] );
+
+			remove_action( 'woocommerce_subscription_status_on-hold_to_active', [ $this->subscription_service, 'reactivate_subscription' ] );
```

Comment:
I've added this to prevent the WCPay Server API call which attempts to reactive the subscription.

This is unnecessary as the WCPay subscription is already active and does not need to be reactivated.

Reply:
Might be worth commenting that in code?

Reply:
Code comment added in 57c3d1c9eb6f0ecce74d0c0969b8c3eb6a7acf61